archive/100-theorems-list/16_abel_ruffini.lean

```diff
@@ -0,0 +1,222 @@
+/-
+Copyright (c) 2021 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import field_theory.abel_ruffini
+import analysis.calculus.local_extr
+/-!
+Construction of an algebraic number that is not solvable by radicals.
+
+The main ingredients are:
+ * `solvable_by_rad.is_solvable'` in `field_theory/abel_ruffini` :
+  an irreducible polynomial with an `is_solvable_by_rad` root has solvable Galois group
+ * `gal_action_hom_bijective_of_prime_degree'` in `field_theory/polynomial_galois_group` :
+  an irreducible polynomial of prime degree with 1-3 non-real roots has full Galois group
+ * `equiv.perm.not_solvable` in `group_theory/solvable` : the symmetric group is not solvable
+
+Then all that remains is the construction of a specific polynomial satisfying the conditions of
+  `gal_action_hom_bijective_of_prime_degree'`, which is done in this file.
+
+-/
+
+namespace abel_ruffini
+
+open function polynomial polynomial.gal
+
+local attribute [instance] splits_ℚ_ℂ
+
+variables (R : Type*) [comm_ring R] (a b : ℕ)
+
+/-- A quintic polynomial that we will show is irreducible -/
+noncomputable def Φ : polynomial R := X ^ 5 - C ↑a * X + C ↑b
+
+variables {R}
+
+@[simp] lemma map_Phi {S : Type*} [comm_ring S] (f : R →+* S) : (Φ R a b).map f = Φ S a b :=
+by simp [Φ]
+
+@[simp] lemma coeff_zero_Phi : (Φ R a b).coeff 0 = ↑b :=
+begin
+  rw [Φ, coeff_add, coeff_sub, coeff_X_pow, coeff_C_mul, coeff_X, coeff_C],
+  norm_num,
+end
```

Comment:
```suggestion
by simp [Φ, coeff_X_pow]
```

Reply:
This is shorter, but is it necessarily better? I have two concerns: it's slower, and it doesn't generalize to other coefficients. @jcommelin what do you think? (and same for the next suggestion)

Reply:
Is it really slower? I don't see how my suggestion could be more expensive than what you already had.

Reply:
And I'm not sure what you mean by it not generalizing to other coefficients.

Reply:
It looks fine to me. @tb65536 can you explain what your concerns are? The speed could be solved by squeezing the simp.

Reply:
I'm happy with the suggestion if you think that it's fine. It is a bit slower (the simp seems to take more time than the norm_num), and the original proof works verbatim for other coefficients, whereas the new one doesn't.